docs/modules/ROOT/pages/design/database-migrations.adoc

```diff
@@ -0,0 +1,56 @@
+= Database Migrations
+
+This document details the database migration strategy KubeArchive uses. The main goals
+of the migrations are:
+
+* Make the upgrade experience easier
+* Grant the ability of undo DB changes if something goes wrong
+
+**Note**: the implementation of database migrations do not include the execution of those
+migrations. We believe that production databases should not be modified in an unattended way
+so we leave to the KubeArchive operator the migration executions.
+
+== Requirements
+
+* Migrations are run by the PostgreSQL user that KubeArchive uses, to avoid permission problems.
+* Migrations migrate data as well as the schema.
+* Migrations have forward and backwards implementation (`up` and `down` files).
+
+== Implementation
+
+We decided to use the
+link:https://github.com/golang-migrate/migrate[`migrate` tool]
+due to its popularity and because its maintained. This tool automates certain
+processes related with migrations, but its a thin layer that can be easily understood
+and replaced if needed.
+
+The migrations files are provided for each database KubeArchive supports within its integration
+folder (`integrations/database/<db-name>/migrations`. Each migration has two files: `up` and `down`,
+which enables rollbacks. The `up` file applies changes in tables and data and the `down` files undoes
+those changes. Each file should start with a two digit number and an underscore (for example
+`01_initial.up.sql` and `01_initial.down.sql`).
+
+== Execution
+
+To apply migrations with `migrate` the command line looks like this:
+
+[source,bash]
+----
+migrate \
+    -path migrations/ \
+    -database <driver>://<username>:<password>@<ip/dns>:<port>/<database> \
+    up
+----
```

Comment:
Is there a way to know how many migrations are we applying with this command so we know how many we should downgrade in a rollback?

Also, what happens if one of the migration fails? It rollbacks to the initial state?